### PR TITLE
Add ability to assign arbitrary segment ID to /display_ads API

### LINKED ISSUE #19496: Add ability to assign arbitrary segment ID to /display_ads API
## Is this feature related to a problem? Please describe.

The current `/display_ads/` API accepts `audience_segment_type`, which is a nice convenient way to set a segment in the "automated segments" camp — because these are one per type and it's nice to not have to know the ID.

However, for manual segments, which we are working on the creation of via API, we'll want to take the segment and assign it to a billboard (currently named display ad)

## Describe the solution you’d like

In addition to the `audience_segment_type`, I should also be able to designate a `audience_segment_id` which could be a manual segment.

- A `audience_segment_id` can be sent without a `audience_segment_type`.
- If a `audience_segment_type` is sent that does not match with the `audience_segment_id`, we should send `422 (Unprocessable Content)`.
- `audience_segment_type` can be sent without an ID, except for type of "manual" because that could be many different segments.

## Definition of Done

- [ ] attribute added to endpoint and it sets the proper value on the billboard itself.

### PR DESCRIPTION
**Pull request by jaw6**
## What type of PR is this? (check all applicable)

- [ ] Refactor
- [x] Feature
- [ ] Bug Fix
- [ ] Optimization
- [ ] Documentation Update

## Description

> In addition to the `audience_segment_type`, I should also be able to designate a `audience_segment_id` which could be a manual segment.
> 
> - A `audience_segment_id` can be sent without a `audience_segment_type`.
> - If a `audience_segment_type` is sent that does not match with the `audience_segment_id`, we should send `422 (Unprocessable Content)`.
> - `audience_segment_type` can be sent without an ID, except for type of "manual" because that could be many different segments.

## Related Tickets & Documents

- Closes #19496 

## Added/updated tests?

- [x] Yes
- [ ] No, and this is why: _please replace this line with details on why tests
      have not been included_
- [ ] I need help with writing tests

## [optional] What gif best describes this PR or how it makes you feel?

alt_text

### CODE REVIEW

File: app/models/display_ad.rb [hunk 1]
```diff
@@ -187,4 +189,22 @@ def should_refresh_audience_segment?
       audience_segment &&
       audience_segment.updated_at < 1.day.ago
   end
+
+  def valid_audience_segment_type
+    return if audience_segment_type.blank?
+
+    errors.add(:audience_segment_type) unless AudienceSegment.valid_type?(audience_segment_type)
+  end
```
Comment (filleduchaos) [inline app/models/display_ad.rb:197]: I'm fairly sure it's possible to express this as an `inclusion` validation, which would have Rails provide a nice error message

(Also would be nice to have readable error messages for the other validations)

**Review by filleduchaos — COMMENTED** [review]
@jaw6 This mostly checks out 🥳 but it feels a bit off that the `audience_segment_type` may not be populated since it (now) gets serialized - it's possible to fetch a billboard that's linked to a segment and get a response that says its segment type is null (granted, the segment ID would be present, which makes this more of a nitpick, but still). Maybe falling back to the associated audience segment's type would help?

**Review by rt4914 — COMMENTED** [review]
The code looks good. Not very confident about reviewing the `display_ad.rb` related changes.

**Review by filleduchaos — APPROVED** [review]
🎉  itsallcomingtogether.gif

**Pull request merged**